digdag-core/src/main/java/io/digdag/core/agent/DefaultSecretProvider.java

```diff
@@ -46,22 +52,20 @@
         Preconditions.checkArgument(key.indexOf('*') == -1, errorMessage);
 
         //// Secret access control:
-        // 1. Reject if the operator doesn't need the access to the secret (operatorSecretFilter).
-        // 2. Allow if users explicitly grant access using _secret directive (Config grants).
-        // 3. Allow if system access policy (SecretAccessPolicy) allows.
-        // 4. Reject.
+        // 1. If users explicitly grant access using _secret directive in workflow definition (Config grants):
+        //    1-a. Allow if the operator wants to use it (operatorSecretFilter)
+        //    1-b. Reject.
+        // 2. If the system access policy file (SecretAccessPolicy) allows:
+        //    2-a. Allow if the operator wants to use it (operatorSecretFilter)
```

Comment:
1-a and 2-a is actually different in terms of existence of user grant. Could you add a bit detailed explanations?